project/Sheng08/README.md

```diff
@@ -0,0 +1,381 @@
+        - parameters:
+            obj_file (string): .obj file path
+
+    extractOBB
+        Extract the oriented bounding box for the provided mesh data
+        - return:
+            (vector<glm::vec3>): Return 8 corner vertices of OBB
+
+    displayInteractiveView
+        Visualize and interact with the mesh and OBB
+        - parameters:
+            show_original_mesh (bool): Whether to show the original mesh or not
+            display_obb (bool): Whether to display the OBB or not
+    ```
+
+**For Python**:
+
+Import `_meshdatahandler`, `_obb_extractor` and `_obb_interface` for `Python` APIs.
+
+* `MeshDataHandler` class:
+    - `mesh_data = MeshDataHandler(obj_file_path)`: Initializes MeshData by
+      reading `.obj` files.
+    - `mesh_data.getVertices()`: Returns a list of vertices.
+    - `mesh_data.getEdges()`: Returns a list of edges.
+    - `mesh_data.getFaces()`: Returns a list of faces.
+
+* `OBBExtractor` class:
+    - `obb_extractor = OBBExtractor(mesh_data)`: Uses MeshDataHandler class to
+      extract Oriented Bounding Boxes (OBB).
+    - `obb_extractor.computeOBB()`: Computes the OBB using Principal Component
+      Analysis (PCA) and returns 8 corner vertices of OBB.
+    - `obb_extractor.getMinimumVolume()`: Returns the volume of the OBB.
+    - `obb_extractor.getOptimalDirection()`: Returns the optimal direction vector.
+    - `obb_extractor.getOrientation()`: Returns the orientation of the OBB.
+
+* `OOBInterface` class:
+    - `oob_interface = OOBInterface()`: Initializes the interface.
+    - `oob_interface.loadMeshData(obj_file_path)`: Loads the mesh data into the interface.
+    - `oob_interface.extractOBB()`: Extracts the oriented bounding box for the
+      provided mesh data and returns 8 corner vertices of OBB.
+    - `oob_interface.displayInteractiveView(show_original_mesh=True, display_obb=True)`:
+      Visualizes and interacts with the mesh and OBB. The user can specify whether
+      toshow the original mesh and/or the OBB.
+
+## Engineering Infrastructure
+
+- **Automatic Build System**:
+    - [GNU make](https://www.gnu.org/software/make/manual/make.html)
+    - [CMake](https://cmake.org/)
+    - [Docker](https://www.docker.com)
+- **Version Control**:
+    - Git
+    - Github
+- **Testing Framework**:
+    - C++: [Google Test](https://github.com/google/googletest)
+    - Python: [pytest](https://docs.pytest.org/en/7.4.x/)
+- **Documentation**:
+    - Markdown
+    - (Optional) C++: [Doxygen](https://www.doxygen.nl/)
+    - (Optional) Python: [Sphinx](https://www.sphinx-doc.org/en/master/), [pydoc](https://docs.python.org/zh-tw/3/library/pydoc.html)
+- **Continuous Integration**
+  - [Github Actions](https://github.com/features/actions)
+
+## Schedule
+
+Week 1 (11/6):
+- Initial Setup and Environment Configuration
+  - Define project scope for OrientMesh Toolkit.
+  - Set up GitHub repository and version control with Git.
+  - Install the required environment (C++, Python, `pybind11`, `.obj` file 
```

Comment:
You will need a dependency manager or it will hard to get things reproducible.  For a term project I don't mind things get irreproducible.  In real engineering it is not acceptable though.